### PR TITLE
Add JSON web ACL rule attribute

### PR DESCRIPTION
**Pull request by nam054**
### Description

Add `rule_json` attribute to allow for raw JSON input for rules.

### Relations

Relates #35143

### Output from Acceptance Testing

```console
> make testacc TESTARGS='-run=TestAccWAFV2WebACL_' PKG=wafv2
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.5 test ./internal/service/wafv2/... -v -count 1 -parallel 20  -run=TestAccWAFV2WebACL_ -timeout 360m
=== RUN   TestAccWAFV2WebACL_basic
=== PAUSE TestAccWAFV2WebACL_basic
=== RUN   TestAccWAFV2WebACL_Update_rule
=== PAUSE TestAccWAFV2WebACL_Update_rule
=== RUN   TestAccWAFV2WebACL_Update_ruleProperties
=== PAUSE TestAccWAFV2WebACL_Update_ruleProperties
=== RUN   TestAccWAFV2WebACL_Update_nameForceNew
=== PAUSE TestAccWAFV2WebACL_Update_nameForceNew
=== RUN   TestAccWAFV2WebACL_disappears
=== PAUSE TestAccWAFV2WebACL_disappears
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_basic
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_basic
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ATPRuleSet
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ATPRuleSet
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl
=== RUN   TestAccWAFV2WebACL_ManagedRuleGroup_specifyVersion
=== PAUSE TestAccWAFV2WebACL_ManagedRuleGroup_specifyVersion
=== RUN   TestAccWAFV2WebACL_minimal
=== PAUSE TestAccWAFV2WebACL_minimal
=== RUN   TestAccWAFV2WebACL_RateBased_basic
=== PAUSE TestAccWAFV2WebACL_RateBased_basic
=== RUN   TestAccWAFV2WebACL_ByteMatchStatement_basic
=== PAUSE TestAccWAFV2WebACL_ByteMatchStatement_basic
=== RUN   TestAccWAFV2WebACL_ByteMatchStatement_ja3fingerprint
=== PAUSE TestAccWAFV2WebACL_ByteMatchStatement_ja3fingerprint
=== RUN   TestAccWAFV2WebACL_ByteMatchStatement_jsonBody
=== PAUSE TestAccWAFV2WebACL_ByteMatchStatement_jsonBody
=== RUN   TestAccWAFV2WebACL_ByteMatchStatement_body
=== PAUSE TestAccWAFV2WebACL_ByteMatchStatement_body
=== RUN   TestAccWAFV2WebACL_ByteMatchStatement_headerOrder
=== PAUSE TestAccWAFV2WebACL_ByteMatchStatement_headerOrder
=== RUN   TestAccWAFV2WebACL_GeoMatch_basic
=== PAUSE TestAccWAFV2WebACL_GeoMatch_basic
=== RUN   TestAccWAFV2WebACL_GeoMatch_forwardedIP
=== PAUSE TestAccWAFV2WebACL_GeoMatch_forwardedIP
=== RUN   TestAccWAFV2WebACL_LabelMatchStatement
=== PAUSE TestAccWAFV2WebACL_LabelMatchStatement
=== RUN   TestAccWAFV2WebACL_RuleLabels
=== PAUSE TestAccWAFV2WebACL_RuleLabels
=== RUN   TestAccWAFV2WebACL_IPSetReference_basic
=== PAUSE TestAccWAFV2WebACL_IPSetReference_basic
=== RUN   TestAccWAFV2WebACL_IPSetReference_forwardedIP
=== PAUSE TestAccWAFV2WebACL_IPSetReference_forwardedIP
=== RUN   TestAccWAFV2WebACL_RateBased_customKeys
=== PAUSE TestAccWAFV2WebACL_RateBased_customKeys
=== RUN   TestAccWAFV2WebACL_RateBased_forwardedIP
=== PAUSE TestAccWAFV2WebACL_RateBased_forwardedIP
=== RUN   TestAccWAFV2WebACL_RuleGroupReference_basic
=== PAUSE TestAccWAFV2WebACL_RuleGroupReference_basic
=== RUN   TestAccWAFV2WebACL_RuleGroupReference_shieldMitigation
=== PAUSE TestAccWAFV2WebACL_RuleGroupReference_shieldMitigation
=== RUN   TestAccWAFV2WebACL_RuleGroupReference_manageShieldMitigationRule
=== PAUSE TestAccWAFV2WebACL_RuleGroupReference_manageShieldMitigationRule
=== RUN   TestAccWAFV2WebACL_Custom_requestHandling
=== PAUSE TestAccWAFV2WebACL_Custom_requestHandling
=== RUN   TestAccWAFV2WebACL_Custom_response
=== PAUSE TestAccWAFV2WebACL_Custom_response
=== RUN   TestAccWAFV2WebACL_tags
=== PAUSE TestAccWAFV2WebACL_tags
=== RUN   TestAccWAFV2WebACL_RateBased_maxNested
=== PAUSE TestAccWAFV2WebACL_RateBased_maxNested
=== RUN   TestAccWAFV2WebACL_Operators_maxNested
=== PAUSE TestAccWAFV2WebACL_Operators_maxNested
=== RUN   TestAccWAFV2WebACL_tokenDomains
=== PAUSE TestAccWAFV2WebACL_tokenDomains
=== RUN   TestAccWAFV2WebACL_associationConfigCloudFront
=== PAUSE TestAccWAFV2WebACL_associationConfigCloudFront
=== RUN   TestAccWAFV2WebACL_associationConfigRegional
=== PAUSE TestAccWAFV2WebACL_associationConfigRegional
=== RUN   TestAccWAFV2WebACL_CloudFrontScope
=== PAUSE TestAccWAFV2WebACL_CloudFrontScope
=== RUN   TestAccWAFV2WebACL_ruleJSON
=== PAUSE TestAccWAFV2WebACL_ruleJSON
=== CONT  TestAccWAFV2WebACL_basic
=== CONT  TestAccWAFV2WebACL_LabelMatchStatement
=== CONT  TestAccWAFV2WebACL_Custom_response
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_specifyVersion
=== CONT  TestAccWAFV2WebACL_RateBased_forwardedIP
=== CONT  TestAccWAFV2WebACL_Custom_requestHandling
=== CONT  TestAccWAFV2WebACL_IPSetReference_forwardedIP
=== CONT  TestAccWAFV2WebACL_RateBased_customKeys
=== CONT  TestAccWAFV2WebACL_RuleGroupReference_shieldMitigation
=== CONT  TestAccWAFV2WebACL_IPSetReference_basic
=== CONT  TestAccWAFV2WebACL_GeoMatch_forwardedIP
=== CONT  TestAccWAFV2WebACL_RuleGroupReference_manageShieldMitigationRule
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_headerOrder
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_body
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_jsonBody
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_ja3fingerprint
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_basic
=== CONT  TestAccWAFV2WebACL_RateBased_basic
=== CONT  TestAccWAFV2WebACL_minimal
=== CONT  TestAccWAFV2WebACL_GeoMatch_basic
--- PASS: TestAccWAFV2WebACL_minimal (37.97s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_basic
--- PASS: TestAccWAFV2WebACL_basic (42.97s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl
--- PASS: TestAccWAFV2WebACL_IPSetReference_basic (73.24s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ATPRuleSet
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_body (91.65s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_specifyVersion (91.83s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl (55.65s)
=== CONT  TestAccWAFV2WebACL_associationConfigCloudFront
--- PASS: TestAccWAFV2WebACL_RateBased_forwardedIP (104.02s)
=== CONT  TestAccWAFV2WebACL_ruleJSON
--- PASS: TestAccWAFV2WebACL_GeoMatch_basic (113.42s)
=== CONT  TestAccWAFV2WebACL_CloudFrontScope
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_headerOrder (114.77s)
=== CONT  TestAccWAFV2WebACL_associationConfigRegional
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_jsonBody (118.78s)
=== CONT  TestAccWAFV2WebACL_RuleLabels
--- PASS: TestAccWAFV2WebACL_RateBased_basic (135.37s)
=== CONT  TestAccWAFV2WebACL_Operators_maxNested
--- PASS: TestAccWAFV2WebACL_LabelMatchStatement (135.63s)
=== CONT  TestAccWAFV2WebACL_tokenDomains
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_basic (135.96s)
=== CONT  TestAccWAFV2WebACL_Update_nameForceNew
--- PASS: TestAccWAFV2WebACL_GeoMatch_forwardedIP (142.99s)
=== CONT  TestAccWAFV2WebACL_disappears
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_ja3fingerprint (143.45s)
=== CONT  TestAccWAFV2WebACL_RuleGroupReference_basic
--- PASS: TestAccWAFV2WebACL_associationConfigCloudFront (69.57s)
=== CONT  TestAccWAFV2WebACL_RateBased_maxNested
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_shieldMitigation (170.65s)
=== CONT  TestAccWAFV2WebACL_Update_ruleProperties
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_manageShieldMitigationRule (173.85s)
=== CONT  TestAccWAFV2WebACL_tags
--- PASS: TestAccWAFV2WebACL_ruleJSON (74.32s)
=== CONT  TestAccWAFV2WebACL_Update_rule
--- PASS: TestAccWAFV2WebACL_CloudFrontScope (66.90s)
--- PASS: TestAccWAFV2WebACL_associationConfigRegional (66.06s)
--- PASS: TestAccWAFV2WebACL_IPSetReference_forwardedIP (191.07s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ATPRuleSet (120.46s)
--- PASS: TestAccWAFV2WebACL_tokenDomains (63.75s)
--- PASS: TestAccWAFV2WebACL_disappears (57.49s)
--- PASS: TestAccWAFV2WebACL_Operators_maxNested (66.49s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet (111.13s)
--- PASS: TestAccWAFV2WebACL_Custom_response (204.92s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig (114.13s)
--- PASS: TestAccWAFV2WebACL_Custom_requestHandling (215.09s)
--- PASS: TestAccWAFV2WebACL_RuleLabels (98.65s)
--- PASS: TestAccWAFV2WebACL_RateBased_maxNested (50.44s)
--- PASS: TestAccWAFV2WebACL_Update_nameForceNew (82.87s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_basic (181.18s)
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_basic (91.94s)
--- PASS: TestAccWAFV2WebACL_Update_rule (59.98s)
--- PASS: TestAccWAFV2WebACL_tags (73.85s)
--- PASS: TestAccWAFV2WebACL_Update_ruleProperties (81.43s)
--- PASS: TestAccWAFV2WebACL_RateBased_customKeys (276.99s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/wafv2      282.915s
```

### CODE REVIEW

File: internal/service/wafv2/web_acl.go [hunk 1]
```diff
@@ -14,6 +14,7 @@ import (
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/wafv2"
 	awstypes "github.com/aws/aws-sdk-go-v2/service/wafv2/types"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/structure"
```
Comment (ewbankkit) [inline internal/service/wafv2/web_acl.go:17]: Should be `github.com/hashicorp/terraform-plugin-sdk/v2/helper/structure` or else we're pulling in the _very_ old Plugin SDK v1.

File: website/docs/r/wafv2_web_acl.html.markdown [hunk 2]
```diff
@@ -465,6 +465,7 @@ This resource supports the following arguments:
 * `description` - (Optional) Friendly description of the WebACL.
 * `name` - (Required, Forces new resource) Friendly name of the WebACL.
 * `rule` - (Optional) Rule blocks used to identify the web requests that you want to `allow`, `block`, or `count`. See [`rule`](#rule-block) below for details.
+* `rule_json` (Optional) Raw JSON block to allow more than three rules. Conflicts with `rule` attribute. **NOTE There is no drift detection at this time. If you use this attribute instead of `rule`, you will be foregoing drift detection.
```
Comment (ewbankkit) [inline website/docs/r/wafv2_web_acl.html.markdown:468]: Maybe mention that this attribute is for advanced use cases when there are more than 3 levels of nested statements.
Also, can we provide a link to the JSON structure, e.g. https://docs.aws.amazon.com/waf/latest/APIReference/API_CreateWebACL.html?

**Review by jar-b — APPROVED** [review]
LGTM 🎉 

```console
% make testacc PKG=wafv2 TESTS=TestAccWAFV2WebACL_
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.22.5 test ./internal/service/wafv2/... -v -count 1 -parallel 20 -run='TestAccWAFV2WebACL_'  -timeout 360m

=== NAME  TestAccWAFV2WebACL_associationConfigCloudFront
    web_acl_test.go:2907: skipping tests; AWS_DEFAULT_REGION (us-west-2) not supported. Supported: [us-east-1]
--- SKIP: TestAccWAFV2WebACL_associationConfigCloudFront (0.47s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet
=== NAME  TestAccWAFV2WebACL_CloudFrontScope
    web_acl_test.go:3004: skipping tests; AWS_DEFAULT_REGION (us-west-2) not supported. Supported: [us-east-1]
--- SKIP: TestAccWAFV2WebACL_CloudFrontScope (0.47s)
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_body
--- PASS: TestAccWAFV2WebACL_associationConfigRegional (35.89s)
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_headerOrder
--- PASS: TestAccWAFV2WebACL_ruleJSON (58.23s)
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_jsonBody
--- PASS: TestAccWAFV2WebACL_Update_nameForceNew (67.62s)
=== CONT  TestAccWAFV2WebACL_RateBased_customKeys
--- PASS: TestAccWAFV2WebACL_basic (81.69s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_body (91.00s)
=== CONT  TestAccWAFV2WebACL_RateBased_basic
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_specifyVersion (91.53s)
=== CONT  TestAccWAFV2WebACL_ByteMatchStatement_basic
--- PASS: TestAccWAFV2WebACL_GeoMatch_basic (99.16s)
=== CONT  TestAccWAFV2WebACL_disappears
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_ja3fingerprint (101.28s)
=== CONT  TestAccWAFV2WebACL_Update_ruleProperties
--- PASS: TestAccWAFV2WebACL_GeoMatch_forwardedIP (103.82s)
=== CONT  TestAccWAFV2WebACL_IPSetReference_basic
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_basic (116.87s)
=== CONT  TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl
--- PASS: TestAccWAFV2WebACL_Custom_response (126.68s)
=== CONT  TestAccWAFV2WebACL_minimal
--- PASS: TestAccWAFV2WebACL_RateBased_forwardedIP (127.07s)
=== CONT  TestAccWAFV2WebACL_RateBased_maxNested
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_shieldMitigation (127.19s)
=== CONT  TestAccWAFV2WebACL_tokenDomains
--- PASS: TestAccWAFV2WebACL_LabelMatchStatement (128.34s)
=== CONT  TestAccWAFV2WebACL_tags
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ATPRuleSet (132.03s)
=== CONT  TestAccWAFV2WebACL_Update_rule
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_headerOrder (97.70s)
=== CONT  TestAccWAFV2WebACL_RuleLabels
--- PASS: TestAccWAFV2WebACL_disappears (47.68s)
=== CONT  TestAccWAFV2WebACL_Operators_maxNested
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_ACFPRuleSet (149.67s)
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_jsonBody (96.46s)
--- PASS: TestAccWAFV2WebACL_Custom_requestHandling (156.98s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig_BotControl (45.20s)
--- PASS: TestAccWAFV2WebACL_IPSetReference_basic (59.51s)
--- PASS: TestAccWAFV2WebACL_RuleGroupReference_manageShieldMitigationRule (163.83s)
--- PASS: TestAccWAFV2WebACL_IPSetReference_forwardedIP (164.94s)
--- PASS: TestAccWAFV2WebACL_minimal (42.11s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_ManagedRuleGroupConfig (91.97s)
--- PASS: TestAccWAFV2WebACL_tokenDomains (48.93s)
--- PASS: TestAccWAFV2WebACL_RateBased_maxNested (49.21s)
--- PASS: TestAccWAFV2WebACL_ByteMatchStatement_basic (85.14s)
--- PASS: TestAccWAFV2WebACL_RateBased_basic (87.12s)
--- PASS: TestAccWAFV2WebACL_ManagedRuleGroup_basic (179.08s)
--- PASS: TestAccWAFV2WebACL_Operators_maxNested (36.93s)
--- PASS: TestAccWAFV2WebACL_RuleLabels (58.01s)
--- PASS: TestAccWAFV2WebACL_Update_rule (61.74s)
--- PASS: TestAccWAFV2WebACL_Update_ruleProperties (95.91s)
--- PASS: TestAccWAFV2WebACL_tags (74.30s)
--- PASS: TestAccWAFV2WebACL_RateBased_customKeys (196.13s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/wafv2      269.689s
```

File: internal/service/wafv2/flex.go [hunk 3]
```diff
@@ -980,6 +983,22 @@ func expandHeaderMatchPattern(l []interface{}) *awstypes.HeaderMatchPattern {
 	return f
 }
 
+func expandWebACLRulesJSON(rawRules string) ([]awstypes.Rule, error) {
```
[the remark this answers is not in the document]
Comment (jar-b) [reply internal/service/wafv2/flex.go:986]: Added in e461f5331c80135e102b9ff28ea68bf8becaeecc

**Pull request merged**